### PR TITLE
`azurerm_site_recovery_protection_container_mapping` - support `automatic_update_extension.automatic_update_extensions_enabled` and `automatic_update_extension.automation_account_id` properties

### LINKED ISSUE #19407: Support for azurerm_recovery_services_vault Allow Site Recovery to manage Extension update
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Ability to associate an automation account to the ASR allowing the use of automatic extension updates.

### New or Affected Resource(s)/Data Source(s)

azurerm_recovery_services_vault 
azurerm_site_recovery_replicated_vm
### Potential Terraform Configuration

_No response_

### References

https://learn.microsoft.com/en-us/azure/site-recovery/azure-to-azure-autoupdate

### LINKED ISSUE #19691: Support for Automation account Settings in azurerm_site_recovery_replicated_vm
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add the ability for the agent extension to be managed by an atomation account

### New or Affected Resource(s)/Data Source(s)

azurerm_site_recovery_replicated_vm

### Potential Terraform Configuration

```hcl
resource "azurerm_site_recovery_replicated_vm" "protected_vm_tf" {
automation_account_id = var.automation_account_id
}
```

### References

https://github.com/hashicorp/terraform-provider-azurerm/issues/19535

### PR DESCRIPTION
**Pull request by ziyeqf**
closes #19691 
closes #19407 

Test
---
```
TF_ACC=1 go test -v ./internal/services/recoveryservices -run=TestAccSiteRecoveryProtectionContainerMapping -timeout=60m            [ tengzh/issue/vault_auto_update][$!][🐹 v1.19.4][⏱ 22m11s][16:57:14]
=== RUN   TestAccSiteRecoveryProtectionContainerMapping_basic
=== PAUSE TestAccSiteRecoveryProtectionContainerMapping_basic
=== RUN   TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
=== PAUSE TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
=== CONT  TestAccSiteRecoveryProtectionContainerMapping_basic
=== CONT  TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
--- PASS: TestAccSiteRecoveryProtectionContainerMapping_basic (884.46s)
--- PASS: TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension (1268.62s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices      1269.810s
```

### CODE REVIEW

File: internal/services/recoveryservices/site_recovery_protection_container_mapping_resource.go [hunk 1]
```diff
@@ -77,6 +77,18 @@ func resourceSiteRecoveryProtectionContainerMapping() *pluginsdk.Resource {
...
+			"automatic_update_extensions_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"automation_account_id": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				RequiredWith: []string{"automatic_update_extensions_enabled"},
```
Comment (mbfrahry) [inline internal/services/recoveryservices/site_recovery_protection_container_mapping_resource.go:89]: I believe something weird will happen here if they pass in an `automation_account_id` and have `automatic_update_extensions_enabled` set to false. We should just have this wrapped in a TypeList to show that these two attributes go together.

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @ziyeqf, thanks for this PR. It's mostly good but I think that we could lump these two attributes into a TypeList instead.

**Comment by ziyeqf** [comment]
Test after modification
---
```
TF_ACC=1 go test -v ./internal/services/recoveryservices -run=TestAccSiteRecoveryProtectionContainerMapping -timeout=60m
=== RUN   TestAccSiteRecoveryProtectionContainerMapping_basic
=== PAUSE TestAccSiteRecoveryProtectionContainerMapping_basic
=== RUN   TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
=== PAUSE TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
=== CONT  TestAccSiteRecoveryProtectionContainerMapping_basic
=== CONT  TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
--- PASS: TestAccSiteRecoveryProtectionContainerMapping_basic (886.50s)
--- PASS: TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension (1229.09s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices      1230.260s
```

File: internal/services/recoveryservices/site_recovery_protection_container_mapping_resource.go [hunk 2]
```diff
@@ -77,6 +78,28 @@ func resourceSiteRecoveryProtectionContainerMapping() *pluginsdk.Resource {
 				ValidateFunc:     azure.ValidateResourceID,
 				DiffSuppressFunc: suppress.CaseDifference,
 			},
+			"automatic_update_extension_settings": {
```
Comment (katbyte) [inline internal/services/recoveryservices/site_recovery_protection_container_mapping_resource.go:81]: i think settings is redundant?
```suggestion
			"automatic_update_extension": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]
One comment left inline

**Review by mbfrahry — COMMENTED** [review]
Hey @ziyeqf, thanks for making those changes. I was looking more into this and it looks like we are changing the `ProviderSpecificInput` type from the basic type `ReplicationProviderSpecificContainerMappingInput` to `A2A`. I also see that there is another type called `VMwareCbt`.

I don't like that we're favoring one type over the other in this resource and we are making that decision for the user based off their input whilst also not letting them use the other type if they want. 

Do you know if this resource does anything? It doesn't appear so since most of the functionality comes from what you're adding here. If this resource isn't doing anything, it's probably worth creating a new resource and deprecating this one. That new resource will be `azurerm_site_recovery_protection_container_mapping_a2a`. That will also give us the flexibility to add the other type `azurerm_site_recovery_protection_container_mapping_vmware_cbt` if there are asks for it.

**Comment by ziyeqf** [comment]
Hi @mbfrahry , For the current resources `azurerm_site_recovery_xxx`, they are all for `A2A`(Azure To Azure) type. 

For Site Recovery, there are basically four type `A2A`,`V2A(Vmware To Azure) Classic`,`V2A Modern` and `H2A`(HyperV To Azure).  I have worked to add the rest three types and I think the current resources are all for A2A so it should be ok to specify a input type. For the others I'm going to name them as `azure_site_recovery_xxx_v2a_classic` and so on.. 

How do you like that?

**Review by mbfrahry — CHANGES_REQUESTED** [review]
That is fair. So with that being the case, we can nearly revert it back to the way you had it with those attributes being at the top level. But we should also force the type to be `InstanceTypeBasicReplicationProviderSpecificContainerMappingInputInstanceTypeA2A` as in the original PR submitted, the type was being changed back to `ReplicationProviderSpecificContainerMappingInput` if these new attributes weren't set. 

Apologies for the run around on this!

**Comment by ziyeqf** [comment]
temporarily close this PR to reduce noise in refactoring.

**Comment by ziyeqf** [comment]
test

**Comment by katbyte** [comment]
@ziyeqf - is there any progress on getting this merged so we don't need the hack? https://github.com/Azure/azure-rest-api-specs/pull/21652

**Comment by ziyeqf** [comment]
> @ziyeqf - is there any progress on getting this merged so we don't need the hack? [Azure/azure-rest-api-specs#21652](https://github.com/Azure/azure-rest-api-specs/pull/21652)

oh, it's a mistake because Tom has explained it's not a Pandora issue. While the codes on `main` branch has been changed. I'm wondering if we can specify the property type to `A2A` in the create function then we will not need the workaround.

**Comment by ziyeqf** [comment]
new test result
---
```
TF_ACC=1 go test -v ./internal/services/recoveryservices -run=TestAccSiteRecoveryProtectionContainerMapping -timeout=60m
=== RUN   TestAccSiteRecoveryProtectionContainerMapping_basic
=== PAUSE TestAccSiteRecoveryProtectionContainerMapping_basic
=== RUN   TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
=== PAUSE TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
=== CONT  TestAccSiteRecoveryProtectionContainerMapping_basic
=== CONT  TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
--- PASS: TestAccSiteRecoveryProtectionContainerMapping_basic (841.12s)
--- PASS: TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension (1290.34s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices      1291.507s
```

File: internal/services/recoveryservices/site_recovery_protection_container_mapping_resource.go [hunk 3]
```diff
@@ -77,6 +78,28 @@ func resourceSiteRecoveryProtectionContainerMapping() *pluginsdk.Resource {
 				ValidateFunc:     azure.ValidateResourceID,
 				DiffSuppressFunc: suppress.CaseDifference,
 			},
+			"automatic_update_extension": {
```
Comment (katbyte) [inline internal/services/recoveryservices/site_recovery_protection_container_mapping_resource.go:81]: extension is redundant?
```suggestion
			"automatic_update": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by ziyeqf** [comment]
test result after rename
---
```
terraform-provider-azurerm ❯ TF_ACC=1 go test -v ./internal/services/recoveryservices -run=TestAccSiteRecoveryProtectionContainerMapping -timeout=60m
=== RUN   TestAccSiteRecoveryProtectionContainerMapping_basic
=== PAUSE TestAccSiteRecoveryProtectionContainerMapping_basic
=== RUN   TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
=== PAUSE TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
=== CONT  TestAccSiteRecoveryProtectionContainerMapping_basic
=== CONT  TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension
--- PASS: TestAccSiteRecoveryProtectionContainerMapping_basic (848.41s)
--- PASS: TestAccSiteRecoveryProtectionContainerMapping_withAutoUpdateExtension (1179.99s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/recoveryservices      1183.464s
```

**Review by katbyte — APPROVED** [review]
Thanks @ziyeqf - LGTM 🛠️

**Pull request merged**